File: cpp/1944.cpp
#include <iostream>
#include <string>
#include <stack>
#include <algorithm>

using namespace std;

int main()
{
    int n, count = 0;
    string blocks, reversed;
    stack<string> stack;
    cin >> n;
    getchar();

    for (n; n > 0; n--) {
        if (stack.size() == 0) {
            stack.push("F A C E");
        }

        getline(cin, blocks);
        reversed = blocks;
        reverse(reversed.begin(), reversed.end());

        if (reversed == stack.top()) {
            stack.pop();
            count++;
        } else {
            stack.push(blocks);
        }
    }

    cout << count << '\n';
    return 0;
}

File: py/1944.py
def main():
    n = int(input())
    stack = []
    count = 0

    for _ in range(n):
        if len(stack) == 0:
            stack.append(['F', 'A', 'C', 'E'])

        blocks = input().split()

        if blocks[::-1] == stack[-1]:
            stack.pop()
            count += 1
        else:
            stack.append(blocks)

    print(count)


if __name__ == '__main__':
    main()